core/chain-configs/src/genesis_config.rs

```diff
@@ -72,88 +70,143 @@ pub struct GenesisConfig {
-        let mut buf = serde_json::to_vec_pretty(&self.records)
-            .expect("Error serializing the genesis records.");
-        if let Err(err) = file.write_all(&mut buf) {
-            panic!("Failed to write a genesis records file {}", err);
-        }
+impl GenesisRecords {
+    /// Parses GenesisRecords from a JSON string.
+    ///
+    /// It panics if the contents cannot be parsed from JSON to the GenesisConfig structure.
+    pub fn from_json(value: &str) -> Self {
+        serde_json::from_str(value).expect("Failed to deserialize the genesis records.")
+    }
 
-        // Writing config
-        let mut file =
-            File::create(config_path).expect("Failed to create / write a genesis config file.");
-        let mut tmp_records = vec![];
-        swap(&mut tmp_records, &mut self.records);
-        let mut buf =
-            serde_json::to_vec_pretty(self).expect("Error serializing the genesis config.");
-        swap(&mut tmp_records, &mut self.records);
-        if let Err(err) = file.write_all(&mut buf) {
-            panic!("Failed to write a genesis config file {}", err);
-        }
+    /// Reads GenesisRecords from a JSON file.
+    ///
+    /// It panics if file cannot be open or read, or the contents cannot be parsed from JSON to the
+    /// GenesisConfig structure.
+    pub fn from_file<P: AsRef<Path>>(path: P) -> Self {
+        Self::from_json(
+            &std::fs::read_to_string(path).expect("Could not read genesis records file."),
+        )
     }
 
-    pub fn from(config: &str, records_str: Option<String>) -> Self {
-        let mut config: GenesisConfig =
-            serde_json::from_str(config).expect("Failed to deserialize the genesis config.");
-        if config.protocol_version != PROTOCOL_VERSION {
-            panic!(format!(
-                "Incorrect version of genesis config {} expected {}",
-                config.protocol_version, PROTOCOL_VERSION
-            ));
-        }
-        if let Some(records_str) = records_str {
-            let mut records: Vec<StateRecord> = serde_json::from_str(&records_str)
-                .expect("Failed to deserialize the genesis config.");
-            config.records.append(&mut records);
-        }
-        config.init();
-        config
+    /// Writes GenesisRecords to the file.
+    pub fn to_file<P: AsRef<Path>>(&self, path: P) {
+        std::fs::write(
+            path,
+            serde_json::to_vec_pretty(self).expect("Error serializing the genesis records."),
+        )
+        .expect("Failed to create / write a genesis records file.");
+    }
+}
+
+impl Genesis {
+    pub fn new(config: GenesisConfig, records: GenesisRecords) -> Self {
+        let mut genesis = Self { config, records, phantom: PhantomData };
+        genesis.config.total_supply = get_initial_supply(&genesis.records.as_ref());
+        genesis
+    }
+
+    /// Reads Genesis from a single file.
```

Comment:
When I run local node, it still uses the single file setup. Should I fix that?

Reply:
Oh I see. Didn't consider this use case. I guess we should fix it as well.

Reply:
In private discussion with @bowenwang1996 we concluded that this method is not deprecated since we store `~/.near/genesis.json` as a single file containing the config and the records in a single file.